### PR TITLE
Check configmap/secret struct for updating

### PR DESCRIPTION
**Pull request by tomahawk28**
Toward https://github.com/giantswarm/giantswarm/issues/13309

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: service/controller/app/resource/configmap/resource.go [hunk 1]
```diff
@@ -86,7 +87,24 @@ func equals(a, b *corev1.ConfigMap) bool {
...
+			return false
+		}
+
+		err = yaml.Unmarshal([]byte(b.Data["values"]), &dest)
+		if err != nil {
+			return false
+		}
+	}
+
+	if !reflect.DeepEqual(source, dest) {
```
Comment (tomahawk28) [inline service/controller/app/resource/configmap/resource.go:107]: Comparing two values as a struct, not YAML strings would not trigger the constant updates.

**Comment by tomahawk28** [comment]
Tested in `geckon`

**Review by rossf7 — APPROVED** [review]
Nice, good catch. Is there anywhere else we need this? Like in the values service.

LGTM when tested manually and tests are green.

**Comment by tomahawk28** [comment]
Values service requires k8sclient, but we don't need that for map comparison. Let's put those changes into each resource.

**Comment by tomahawk28** [comment]
Tested in `geckon`, no further changes in configmaps, secrets.

**Pull request merged**